lib/inspec/reporters/base.rb

```diff
@@ -5,9 +5,36 @@ class Base
     def initialize(config)
       @config = config
       @run_data = config[:run_data]
+      apply_options_to_run_data
       @output = ""
     end
 
+    # Apply options such as message truncation and removal of backtraces
+    def apply_options_to_run_data
+      message_truncation = @config[:message_truncation] || "ALL"
```

Comment:
Right here, you could call `runtime_config = Inspec::Config.cached` and get an instance of  the runtime config with access to the CLI options or JSON config - no need to sneak it into the per-reporter configs.

Reply:
Thanks, that sounds good! I think that solves the problem Alex mentioned where these options are not working from STDIN config.
It would make it a little odd that the class is instantiated with a `config` hash but we don't use that for config - but it does mean I can revert my changes to the runner.